### ISSUE TITLE
System.ArgumentNullException: Value cannot be null. (Parameter 'second')

### ISSUE THREAD

**Issue by y0uCeF** [DigitalRuby/IPBan#214]
I installed IPBan 1.8.0 using the powershell method. I was mainly interested in the OpenSSH logs from the event viewer. It is the only enabled group in my config file kept as in the default config. No log files are monitored and  appSettings are kept to the default config.

```
<configuration>

  <!-- A collection of log files to parse, these are polled periodically based on PingInterval for each log file -->
  <LogFilesToParse>
    <LogFiles>
    </LogFiles>
  </LogFilesToParse>

  <ExpressionsToBlock>
    <Groups>
      <!--
        This group will block OpenSSH login failures
      -->
      <Group>
        <Source>SSH</Source>
        <Keywords>0x4000000000000000</Keywords>
        <Path>OpenSSH/Operational</Path>
        <FailedLoginThreshold>0</FailedLoginThreshold>
        <Expressions>
          <Expression>
            <XPath>//Data[@Name='payload']</XPath>
            <Regex>
              <![CDATA[
                failed\s+password\s+for\s+(invalid\s+user\s+)?(?<username>[^\s]+)\s+from\s+(?<ipaddress>[^\s]+)\s+port\s+[0-9]+\s+ssh|
                did\s+not\s+receive\s+identification\s+string\s+from\s+(?<ipaddress>[^\s]+)|
                connection\s+closed\s+by\s+((invalid\s+user\s+)?(?<username>[^\s]+)\s+)?(?<ipaddress>[^\s]+)\s+port\s+[0-9]+\s+\[preauth\]|
                disconnected\s+from\s+(invalid\s+user\s+)?(?<username>[^\s]+)\s+(?<ipaddress>[^\s]+)\s+port\s+[0-9]+\s+\[preauth\]|
                disconnected\s+from\s+(?<ipaddress>[^\s]+)\s+port\s+[0-9]+\s+\[preauth\]|
                disconnected\s+from\s+authenticating\s+user\s+(?<username>[^\s]+)\s+(?<ipaddress>[^\s]+)\s+port\s+[0-9]+\s+\[preauth\]
              ]]>
            </Regex>
          </Expression>
        </Expressions>
      </Group>
    </Groups>
  </ExpressionsToBlock>

  <appSettings>
  ...
  </appSettings>
</configuration>
```
I keep getting the exception below
```
2022-09-08 11:45:42.2283|ERROR|IPBan|Error Exception: System.ArgumentNullException: Value cannot be null. (Parameter 'second')
   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument )
   at System.Linq.Enumerable.Union[TSource](IEnumerable`1 , IEnumerable`1 , IEqualityComparer`1 )
   at System.Linq.Enumerable.Union[TSource](IEnumerable`1 , IEnumerable`1 )
   at DigitalRuby.IPBanCore.IPBanConfig.WindowsEventViewerGetGroupsMatchingKeywords(UInt64 keywords)
   at DigitalRuby.IPBanCore.IPBanWindowsEventViewer.ExtractEventViewerXml(XmlDocument doc)
   at DigitalRuby.IPBanCore.IPBanWindowsEventViewer.ProcessEventViewerXml(String xml)
   at DigitalRuby.IPBanCore.IPBanWindowsEventViewer.EventRecordWritten(Object sender, EventRecordWrittenEventArgs e)
2022-09-08 11:45:49.3502|ERROR|IPBan|Error Exception: System.ArgumentNullException: Value cannot be null. (Parameter 'second')
   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument )
   at System.Linq.Enumerable.Union[TSource](IEnumerable`1 , IEnumerable`1 , IEqualityComparer`1 )
   at System.Linq.Enumerable.Union[TSource](IEnumerable`1 , IEnumerable`1 )
   at DigitalRuby.IPBanCore.IPBanConfig.WindowsEventViewerGetGroupsMatchingKeywords(UInt64 keywords)
   at DigitalRuby.IPBanCore.IPBanWindowsEventViewer.ExtractEventViewerXml(XmlDocument doc)
   at DigitalRuby.IPBanCore.IPBanWindowsEventViewer.ProcessEventViewerXml(String xml)
   at DigitalRuby.IPBanCore.IPBanWindowsEventViewer.EventRecordWritten(Object sender, EventRecordWrittenEventArgs e)
```
This was tested on Windows Server 2019 as well as Windows 10.
Am I missing some requirements or is there something to be tweaked in the default config file?

**Comment by jjxtra** [comment]
Fixed in 3ad29f43294e25444b9e573ec62a1761e3297460, please try pulling 1.8.0 release again

**Comment by y0uCeF** [comment]
fixed indeed
